### PR TITLE
5 -  Block_User Controller

### PR DESCRIPTION
**Pull request by TriAnh1995**
- Initializing Block_User Controller

### CODE REVIEW

File: api/internal/controller/block_users.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	}
+	return tx.Transaction(func(*gorm.DB) error {
+		if err = i.checkEmail(ctx, userEmails[0]); err != nil {
+			return err
+		}
+		if err = i.checkEmail(ctx, userEmails[1]); err != nil {
+			return err
+		}
+
+		checkIfFollowed, err := i.repo.CheckIfBlocked(ctx, userEmails)
```
Comment (nhan1603) [inline api/internal/controller/block_users.go]: are you using the wrong function here? if not, please update the variable name for better code understandable

File: api/internal/controller/block_users.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		if err != nil {
+			return ServerError
+		}
+		if checkIfFollowed {
+			return AlreadyBlocked
+		}
+
+		checkIfBlocked, err := i.repo.CheckIfFollowed(ctx, userEmails)
```
Comment (nhan1603) [inline api/internal/controller/block_users.go]: same here

**Review by nhan1603 — CHANGES_REQUESTED** [review]
See comments

File: api/internal/controller/block_users.go (hunk 1 above)
Comment (TriAnh1995) [reply api/internal/controller/block_users.go]: whoops, that my mistake. I will change the variable name.

**Review by nhan1603 — APPROVED** [review]

**Pull request merged**